### DISCUSSION TITLE
Tried to create assistant message with Role.System - think enum is out of date?

### DISCUSSION THREAD

**Discussion by Stiltonator** [RageAgainstThePixel/com.openai.unity#320] [Q&A]
I've tried making a message at the start of my assistant message list with the System role,  but i believe for assistants (perhaps all) you have to now use 'developer'  as system is no longer working.

I am not really super confident programming, i tried creating a custom enum in my project to overwrite your definition but it started getting angry with me saying it caused discrepancies with your one from the .dll  (that all makes sense to me)

Is it possible  (assuming i have identified a real issue)  for me to fix this by myself by... as i say, overwriting your enum declaration to include developer intead of system?

Much appreciated for any help you can give in advance.

**Comment by Stiltonator** [comment]
nevermind i think this is my error.. i dont think im able to set it using system/developer at all,  i was comparing this to the..  standard 'chat' usage/results.

**Reply by StephenHodgson** [reply]
That is correct. `Role.System` is not a valid choice for Assistant endpoints. You can only set `Role.Assistant` or `Role.User`
